Formula/telnet.rb

```diff
@@ -0,0 +1,37 @@
+class Telnet < Formula
+  desc "Apple telnet built from macOS 10.12.4 sources"
+  homepage "https://opensource.apple.com/"
+  url "https://opensource.apple.com/tarballs/remote_cmds/remote_cmds-54.50.1.tar.gz"
+  sha256 "156ddec946c81af1cbbad5cc6e601135245f7300d134a239cda45ff5efd75930"
+
+  depends_on :xcode => :build
+  depends_on :macos => :high_sierra
+
+  resource "libtelnet" do
+    url "https://opensource.apple.com/tarballs/libtelnet/libtelnet-13.tar.gz"
+    sha256 "e7d203083c2d9fa363da4cc4b7377d4a18f8a6f569b9bcf58f97255941a2ebd1"
+  end
+
+  def install
+    resource("libtelnet").stage do
+      xcodebuild "SYMROOT=build"
+
+      libtelnet_dst = buildpath / "telnet.tproj" / "build" / "Products"
+      libtelnet_dst.install("build/Release/libtelnet.a")
+      libtelnet_dst.install("build/Release/usr/local/include/libtelnet/")
+    end
+
+    system "make", "-C", "telnet.tproj", "OBJROOT=build/Intermediates", "SYMROOT=build/Products", "DSTROOT=build/Archive", "CFLAGS=$(CC_Flags) -isystembuild/Products/", "LDFLAGS=$(LD_Flags) -Lbuild/Products/", "install"
+
+    bin.install "telnet.tproj/build/Archive/usr/bin/telnet"
+    man.install "telnet.tproj/build/Archive/usr/share/man/man1/"
+  end
+
+  test do
+    output = shell_output("(sleep 2; echo 'quit') | telnet towel.blinkenlights.nl", 1)
```

Comment:
Can use `pipe_output` here, and you'll want to use `#{bin}/telnet` rather than simply `telnet`.